File: 2103. Rings and Rods/Solution.cpp
#include <algorithm>
#include <string>
using namespace std;

class Solution {
public:
    int countPoints(string rings)
    {
        int cnt[10] = { 0 };
        string colors = "RGB";
        for (size_t i = 0; i < rings.size(); i += 2) {
            int c = colors.find(rings[i]), rod = rings[i + 1] - '0';
            cnt[rod] |= 1 << c;
        }
        return count(begin(cnt), end(cnt), 7);
    }
};

File: 2103. Rings and Rods/Solution.py
class Solution:
    def countPoints(self, rings: str) -> int:
        cnt = [0] * 10
        for color, rod in zip(rings[0::2], rings[1::2]):
            cnt[int(rod)] |= 1 << ("RGB".index(color))
        return sum(e == 7 for e in cnt)
